Dockerfile

```diff
@@ -25,17 +29,33 @@ WORKDIR /app
 COPY . /app
 RUN bundle exec rake install
 
+FROM ruby:3.0-alpine
 
-FROM ruby:2.7-alpine
+ARG UID=${UID:-1000}
+ARG GID=${GID:-1000}
 
 COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
 
-RUN apk add \
-    bash \
-    git \
-    libc6-compat
-
-VOLUME /wiki
 WORKDIR /wiki
+RUN apk add --update \
+            --no-cache \
+            bash \
+            git \
+            shadow \
+            curl \
+            libc6-compat \
+    && rm -rf /var/cache/apk/* \
+    && groupmod -g $GID www-data \
+    && adduser -u $UID -S www-data -G www-data
+
 COPY docker-run.sh /docker-run.sh
+RUN chmod +x /docker-run.sh
+USER www-data
+VOLUME /wiki
+EXPOSE 4567
```

Comment:
Why is the explicit `EXPOSE` required? Does this have implications for the [documented way of running the Docker image](https://github.com/gollum/gollum/wiki/Gollum-via-Docker)?

Reply:
This section is implicit for the HEALTHCHECK. To perform the curl.